src/main/java/BasketInformations.java

```diff
@@ -0,0 +1,98 @@
+package com.company;
+
+import java.util.HashMap;
+
+/**
+ * This class is a basket
+ */
+public class BasketInformations {
+
+	// The product of the basket
+	static HashMap<String, Integer> map = new HashMap<String, Integer>();
+
+	static String[] localProducts = new String[]{"123", "222", "44", "657"};
+
+	// The fact that the basket has promo code
+	private static boolean codeDePromotion = false;
+
+	public void addProductToBasket(String product, Integer price, boolean isPromoCode) {
+		if (isPromoCode) {
+			codeDePromotion = true;
+		} else {
+			map.put(product, price);
+		}
+	}
+
+	public Long getBasketPrice(boolean inCents) {
+		Integer v = 0;
+		for (String s : map.keySet()) {
+			v += map.get(s);
+		}
+		if (codeDePromotion) {
+			v -= 100;
+		}
+		return inCents ? v * 100 : Long.valueOf(v);
+	}
+
+	public void resetBasket() {
+		buyBasket();
+		codeDePromotion = false;
+	}
+
+	public void buyBasket() {
+		map.clear();
+	}
+
+	public boolean isBasketContains(String produit) {
+		boolean found = false;
+		for (String s : map.keySet()) {
+			if (s == produit) found = true;
+		}
+		return found;
+	}
+
+	public void mixWithBasket(BasketInformations b) {
+		map.putAll(b.map);
+	}
+
+	public boolean isSendFree() {
+		boolean free = false;
+
```

Comment:
This function currently looks a little bit difficult of understand. Perhaps you could use the Stream API to find the maximum value.

Here is a suggestion that I haven't test, but will be helpful to get the idea: 

```java
public boolean isSendFree() {
        double total = getBasketPrice(true);
        if (total < 1000) {
            return getTotalPriceForLocalProducts() > 3 || getProductMaxPrice() > 50;
        }
        return true;
    }

    private Double getProductMaxPrice() {
        return CACHE.values().stream()
                .map(Product::getPrice)
                .max(Double::compare)
                .orElse(0D);
    }
````

https://www.baeldung.com/java-collection-min-max